### PR TITLE
Export data and send link via gcloud signed URL

### PR DESCRIPTION
**Pull request by mhfowler**

### CODE REVIEW

File: backend/app/utilities/emailHelper.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+const mustache = require('mustache')
+const path = require('path')
+const fs = require('fs')
+const { PROJECT_DIR } = require('../constants')
+
+
+async function renderTemplateAsEmail(templatePath, emailVars) {
+  const templateDir = path.join(PROJECT_DIR, 'app/templates')
+  const fullTemplatePath = path.join(templateDir, templatePath)
+  const templateData = await fs.readFileSync(fullTemplatePath)
```
Comment (oshoham) [inline backend/app/utilities/emailHelper.js]: do you need an `await` here? I thought readFileSync was synchronous

File: backend/app/utilities/exportData.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  logMessage(`++ exporting data for ${bot}`)
+
+  // create an object with the data we need
+  const userActions = await UserAction.find({bot: bot})
+  const users = await User.find({bot: bot, unsubscribed: false})
+  const campaignActions = await CampaignAction.find({bot: bot})
+  const representatives = await Reps.find({})
+  const committees = await RepresentativeCommittee.find({})
+  const subCommittees = await RepresentativeSubcommittee.find({})
+  // const output = {
```
Comment (oshoham) [inline backend/app/utilities/exportData.js]: can you remove this commented out code?

Comment (mhfowler) [reply backend/app/utilities/exportData.js]: I removed this

File: backend/app/utilities/emailHelper.js (hunk 1 above)
Comment (mhfowler) [reply backend/app/utilities/emailHelper.js]: changed it

File: backend/app/utilities/googleCloud.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+function makeFilePublic(filePath) {
+  let gcs = storage({
+    projectId: process.env.GCLOUD_PROJECT,
+    keyFilename: GCLOUD_KEY_PATH
+  })
+
+  var bucket = gcs.bucket(GCLOUD_BUCKET_NAME)
+
+  // return promise for file upload
+  return new Promise(function(resolve, reject) {
```
Comment (oshoham) [inline backend/app/utilities/googleCloud.js]: you don't really need to use a promise constructor here, you could rewrite this as

```
logMesssage(`++ making ${filePath} public`)
return bucket.file(filePath)
  .makePublic()
  .then(() => {
    logMessage(`gs://${GCLOUD_BUCKET_NAME}/${filePath} is now public.`)
  })
  .catch(err => {
    console.error('ERROR:', err)
    throw err
  })
```

File: backend/app/utilities/googleCloud.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  const expirationDate = now.add(2, 'days')
+  const stringifiedExpirationDate = expirationDate.format('MM-DD-YYYY')
+  logMessage(stringifiedExpirationDate)
+  const signedUrlptions = {
+    action: 'read',
+    expires: stringifiedExpirationDate,
+  }
+
+  // return promise for file upload
+  return new Promise(function(resolve, reject) {
```
Comment (oshoham) [inline backend/app/utilities/googleCloud.js]: also don't need a promise constructor here

File: backend/app/utilities/googleCloud.js (hunk 3 above)
Comment (mhfowler) [reply backend/app/utilities/googleCloud.js]: cool for some reason thought it didn't return a promise before, but seems to be working with this now thanks

**Pull request merged**